merlin/models/tf/models/retrieval.py

```diff
@@ -300,3 +309,277 @@ def YoutubeDNNRetrievalModel(
+        post=post,
+    )
+
+    if not outputs:
+        if not negative_samplers:
+            warnings.warn(
+                "By default, the MatrixFactorization model is trained using constrastive learning."
+                "When `negative_samplers` is not specified, `in-batch` negative sampling is used."
+            )
+            negative_samplers = ["in-batch"]
+        outputs = ContrastiveOutput(
+            to_call=DotProduct(),
+            negative_samplers=negative_samplers,
+            logits_temperature=logits_temperature,
+            schema=candidate,
+            **kwargs,
+        )
+
+    if isinstance(outputs, list):
+        outputs = ParallelBlock(*outputs)
+
+    model = RetrievalModelV2(
+        query=query_encoder,
+        candidate=candidate_encoder,
+        output=outputs,
+    )
+
+    return model
+
+
+def TwoTowerModelV2(
+    query_tower: Encoder,
+    candidate_tower: Encoder,
+    candidate_id_tag=Tags.ITEM_ID,
+    outputs: Optional[Union[ModelOutput, List[ModelOutput]]] = None,
+    logits_temperature: float = 1.0,
+    negative_samplers: ItemSamplersType = None,
+    **kwargs,
+) -> RetrievalModelV2:
+    """Builds the Two-tower architecture, as proposed in [1].
+
+    Example Usage::
+        query = mm.Encoder(user_schema, mm.MLPBlock([128]))
+        candidate = mm.Encoder(item_schema, mm.MLPBlock([128]))
+        model = TwoTowerModel(query, candidate)
+        two_tower.compile(optimizer="adam")
+        two_tower.fit(train_data, epochs=10)
+
+    References
+    ----------
+    [1] Yi, Xinyang, et al.
+        "Sampling-bias-corrected neural modeling for large corpus item recommendations."
+        Proceedings of the 13th ACM Conference on Recommender Systems. 2019.
+
+    Parameters
+    ----------
+    query_tower: Encoder
+        The layer that encodes query features
+    candidate_tower: Encoder
+        The  layer that encodes candidates features
+    candidate_id_tag: Tag, optional
+        The tag to select candidate-id feature, by default `Tags.ITEM_ID`
+    outputs:  Union[ModelOutput, List[ModelOutput]], optional
+        The optional `ModelOutput` or list of `ModelOutput` to apply on the model.
+    logits_temperature: float
+        Parameter used to reduce model overconfidence, so that logits / T.
+        Defaults to 1.
+    negative_samplers: List[ItemSampler]
+        List of samplers for negative sampling, by default None
+
+    Returns
+    -------
+    RetrievalModelV2
+    """
+    assert isinstance(query_tower, Encoder), ValueError(
+        "The query tower should be an instance of `Encoder` class"
+    )
+    assert isinstance(candidate_tower, Encoder), ValueError(
+        "The query tower should be an instance of `Encoder` class"
+    )
+
+    if not outputs:
+        if not negative_samplers:
+            warnings.warn(
+                "By default, the TwoTower model is trained using constrastive learning."
```

Comment:
I think this should not be a warning. For interested uses the docstrings should be enough.